scripts/macos/osxcross-toolchain.cmake

```diff
@@ -0,0 +1,45 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+# Specify default system configuration
+set(CMAKE_SYSTEM_NAME Darwin CACHE STRING "Target operating system name")
+set(CMAKE_SYSTEM_PROCESSOR arm64 CACHE STRING "Target operating system processor")
+set(CMAKE_SYSTEM_VERSION "20.0.0" CACHE STRING "Target operating system version")
+set(CMAKE_CROSSCOMPILING TRUE CACHE BOOL "Target is cross compiled")
+
+if(NOT CMAKE_OSX_ARCHITECTURES)
+    set(CMAKE_OSX_ARCHITECTURES "x86_64;arm64" CACHE STRING "OSX Architectures")
+endif()
+if(NOT CMAKE_OSX_SYSROOT)
+    if(DEFINED ENV{SDKROOT})
+        set(CMAKE_OSX_SYSROOT "$ENV{SDKROOT}")
+    else()
+        message(FATAL_ERROR "One of CMAKE_OSX_SYSROOT or ENV{SDKROOT} must be set")
+    endif()
+endif()
+
+set(OSXCROSS_TARGET_TRIPLE "${CMAKE_SYSTEM_PROCESSOR}-apple-darwin${CMAKE_SYSTEM_VERSION}")
```

Comment:
Is this used?

Reply:
Hopefully, if I got it all cleaned up in this PR, supporting Windows/ARM would just be a matter of writing an appropriate toolchain file, and maybe tweaking the rustlang configuration generated by `__rust_build_toolchain_config`